src/conda_package_handling/conda_fmt.py

```diff
@@ -47,47 +32,69 @@ def supported(fn):
     @staticmethod
     def extract(fn, dest_dir, **kw):
         components = utils.ensure_list(kw.get("components")) or ("info", "pkg")
-        file_id = os.path.basename(fn).replace(".conda", "")
         if not os.path.isabs(fn):
             fn = os.path.normpath(os.path.join(os.getcwd(), fn))
         if not os.path.isdir(dest_dir):
             os.makedirs(dest_dir)
-        for component in components:
-            _extract_component(fn, file_id, component, dest_dir)
+
+        _extract(str(fn), str(dest_dir), components=components)
 
     @staticmethod
     def extract_info(fn, dest_dir=None):
         return CondaFormat_v2.extract(fn, dest_dir, components=["info"])
 
     @staticmethod
-    def create(prefix, file_list, out_fn, out_folder=os.getcwd(), **kw):
+    def create(
+        prefix,
+        file_list,
+        out_fn,
+        out_folder=os.getcwd(),
+        compressor=lambda: zstandard.ZstdCompressor(
+            level=ZSTD_COMPRESS_LEVEL, threads=ZSTD_COMPRESS_THREADS
+        ),
+        **kw,
+    ):
         if os.path.isabs(out_fn):
             out_folder = os.path.dirname(out_fn)
             out_fn = os.path.basename(out_fn)
         conda_pkg_fn = os.path.join(out_folder, out_fn)
-        out_fn = out_fn.replace(".conda", "")
+        file_id = out_fn = out_fn.replace(".conda", "")
         pkg_files = utils.filter_info_files(file_list, prefix)
         info_files = set(file_list) - set(pkg_files)
-        ext, comp_filter, filter_opts = kw.get("compression_tuple") or DEFAULT_COMPRESSION_TUPLE
 
-        with utils.TemporaryDirectory(dir=out_folder) as tmpdir:
-            info_tarball = create_compressed_tarball(
-                prefix, info_files, tmpdir, "info-" + out_fn, ext, comp_filter, filter_opts
-            )
-            pkg_tarball = create_compressed_tarball(
-                prefix, pkg_files, tmpdir, "pkg-" + out_fn, ext, comp_filter, filter_opts
+        # legacy libarchive-ish compatibility
+        ext, comp_filter, filter_opts = compression_tuple = kw.get(
+            "compression_tuple", (None, None, None)
```

Comment:
Shouldn't compression_tuple be documented in the function signature somehow? Otherwise, how we people know this is even an option.  For example, I have no idea what this "filter_opts" variable does later on. I have to first try to parse the code to even begin understanding it.

Reply:
compression_tuple is left over from when the project used libarchive. It is kept for backwards compatibility.

Reply:
compression_tuple was added over summer 2022